core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -149,6 +149,12 @@ public BaseMetastoreCatalogTableBuilder(TableIdentifier identifier, Schema schem
       this.identifier = identifier;
       this.schema = schema;
       this.tableProperties.putAll(tableDefaultProperties());
+
+      // Explicitly set default Parquet compression codecs for new tables
```

Comment:
This seems to only cover catalogs that extend this class.
Is it something we can address in `TableMetadata`?

Reply:
Theroetically we could but it sounds strange to me, making the TableMetadata object construct by default with a few properties set , as its a public class?

Reply:
Well, the problem here is that we update multiple places with the same logic and it only covers some implementations. I am not sure how I feel about changing `TableMetadata` either.

Reply:
Took a look, REST catalog table builder doesnt seem to build TableMetadata themselves (gets from the response object).  So this wont help that case.  I tried to organize the code a little bit into TableMetadata class in any case